File: .sdk-repos/orchestration-cluster-api-csharp/examples/Incident.cs
// Compilable usage examples for incident operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class IncidentExamples
{
    #region GetIncident
    // <GetIncident>
    public static async Task GetIncidentExample(IncidentKey incidentKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetIncidentAsync(incidentKey);
        Console.WriteLine($"Incident: {result.IncidentKey}");
    }
    // </GetIncident>
    #endregion GetIncident

    #region ResolveIncident

    // <ResolveIncident>
    public static async Task ResolveIncidentExample(IncidentKey incidentKey)
    {
        using var client = CamundaClient.Create();

        await client.ResolveIncidentAsync(
            incidentKey,
            new IncidentResolutionRequest());
    }
    // </ResolveIncident>
    #endregion ResolveIncident

    #region SearchIncidents

    // <SearchIncidents>
    public static async Task SearchIncidentsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchIncidentsAsync(new IncidentSearchQuery());

        foreach (var incident in result.Items)
        {
            Console.WriteLine($"Incident: {incident.IncidentKey}");
        }
    }
    // </SearchIncidents>
    #endregion SearchIncidents
}

File: .sdk-repos/version-8.9/orchestration-cluster-api-python/examples/incident.py
# Compilable usage examples for incident management.
# These examples are type-checked during build to guard against API regressions.
from __future__ import annotations

from camunda_orchestration_sdk import (
    CamundaClient,
    IncidentKey,
    IncidentSearchQuery,
    Unset,
)


# region SearchIncidents
def search_incidents_example() -> None:
    client = CamundaClient()

    result = client.search_incidents(
        data=IncidentSearchQuery()
    )

    if not isinstance(result.items, Unset):
        for incident in result.items:
            print(f"Incident key: {incident.incident_key}")
# endregion SearchIncidents


# region GetIncident
def get_incident_example(incident_key: IncidentKey) -> None:
    client = CamundaClient()

    incident = client.get_incident(incident_key=incident_key)

    print(f"Incident error type: {incident.error_type}")
# endregion GetIncident


# region ResolveIncident
def resolve_incident_example(incident_key: IncidentKey) -> None:
    client = CamundaClient()

    client.resolve_incident(incident_key=incident_key)
# endregion ResolveIncident
